golem_messages/message/concents.py

```diff
@@ -609,3 +610,105 @@ class ClientAuthorization(base.Message):
     __slots__ = [
         'client_public_key',
     ] + base.Message.__slots__
+
+
+class NonceAbstractMessage(base.Message):
+    """
+    Abstract message containing `nonce` field and its validation.
+    """
+
+    __slots__ = [
+        'nonce',
+    ] + base.Message.__slots__
+
+    def deserialize_slot(self, key, value):
+        value = super().deserialize_slot(key, value)
+        if key == 'nonce':
+            validators.validate_integer(field_name=key, value=value)
+        return value
+
+
+class TransactionSigningRequest(NonceAbstractMessage):
+    """
+    Message sent from SCI transaction signing callback to a Middleman,
+    containing data about transaction which client wants to sign using SigningService.
+    """
+
+    TYPE = CONCENT_MSG_BASE + 23
+
+    __slots__ = [
+        'gasprice',
+        'startgas',
+        'to',
+        'value',
+        'data',
+    ] + NonceAbstractMessage.__slots__
+
+    def deserialize_slot(self, key, value):
+        value = super().deserialize_slot(key, value)
+        if key == 'to':
+            validators.validate_varchar(
+                field_name=key,
+                value=value,
+                max_length=20,
+            )
+        if key == 'data':
+            validators.validate_bytes(
+                field_name=key,
+                value=value,
+            )
+        if key in ('gasprice', 'startgas', 'value'):
+            validators.validate_integer(field_name=key, value=value)
+        return value
+
+
+class SignedTransaction(TransactionSigningRequest):
+    """
+    Message sent from SigningService to the Middleman,
+    if transaction was successfully signed,
+    containing data about transaction and its signature.
+
+    Middleman should copy the signature data to the transaction object passed
+    to the callback by SCI.
+    """
+
+    TYPE = CONCENT_MSG_BASE + 24
+
+    __slots__ = [
+        'v',
+        'r',
+        's',
+    ] + TransactionSigningRequest.__slots__
+
+    def deserialize_slot(self, key, value):
+        value = super().deserialize_slot(key, value)
+        if key in ('v', 'r', 's'):
+            validators.validate_integer(field_name=key, value=value)
+        return value
+
+
+class TransactionRejected(NonceAbstractMessage):
+    """
+    Message sent from SigningService to the Middleman,
+    if transaction cannot be signed from any of various reasons.
+    """
+
+    TYPE = CONCENT_MSG_BASE + 25
+
+    @enum.unique
+    class TransactionRejectionReason(enum.Enum):
```

Comment:
If it's nested in another class then it would be better to shorten it to just `Reason`.

Reply:
So I will follow up `golem-messages` convention and make it a `REASON`

Reply:
:+1: done